### ISSUE TITLE
Rule Request: don't allow numbers in code

### ISSUE THREAD

**Issue by yonat** [realm/SwiftLint#2576]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

> 1. Why should this rule be added? 

Raw numbers should be put in constants, not be part of complex code.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Triggering:

```swift
NSLayoutConstraint(
    item: view.mainMessageLabel, attribute: .centerY, relatedBy: .equal,
    toItem: view, attribute: .centerY, multiplier: (((216 + 31) / 333) * 2), constant: 0.0
)

return 62.0
```

(taken from real world code, I kid you not.)

Non-triggering:

```swift
let ratio: CGFloat = (((216 + 31) / 333) * 2)
let padding = 62.0
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

Not sure. Currently I use the following custom rule:

```
  numbers_smell:
    regex: "(return |case |\\w\\(|: |\\?\\? |\\, |== |<=? |>=? |\\+= |\\-= |\\/= |\\*= |%= |\\w\\.\\w+ = )\\(*\\d{2,}"
    message: "Numbers smell; define a constant instead."
```

> 4. Should the rule be opt-in or enabled by default? Why?

Enabled by default, I think there's a general consensus about numbers in code.

**Comment by akshitzaveri** [comment]
Hi. @yonat Does this rule works for negative numbers?

**Comment by akshitzaveri** [comment]
![Screenshot 2020-03-23 at 14 50 03](https://user-images.githubusercontent.com/6559898/77301416-a51b6c00-6d15-11ea-83a3-fb1735e6db9c.png)

**Comment by yonat** [comment]
You're right, here is a modification that picks up (most) negative numbers:

```yaml
    regex: "(return |case |\\w\\(|: |\\?\\? |\\, |== |<=? |>=? |\\+= |\\-= |\\/= |\\*= |%= |\\w\\.\\w+ = )\\(*-?\\d{2,}"
```

**Comment by akshitzaveri** [comment]
Thanks @yonat 
I will try this. I wish I knew enough regex to fix this issue by myself.
